core/shared/src/main/scala/cats/effect/Resource.scala

```diff
@@ -561,14 +561,33 @@ abstract private[effect] class ResourceInstances0 {
       def F = F0
     }
 
-  implicit def catsEffectSemigroupKForResource[F[_], A](implicit F0: Monad[F],
-                                                        K0: SemigroupK[F]): ResourceSemigroupK[F] =
+  implicit def catsEffectSemigroupKForResource2[F[_], E](implicit F0: MonadError[F, E]): SemigroupK[Resource[F, *]] =
+    new SemigroupK[Resource[F, *]] {
+      final override def combineK[A](a: Resource[F, A], b: Resource[F, A]): Resource[F, A] =
+        MonadError[Resource[F, *], E].handleErrorWith(a)(_ => b)
+    }
+
+  // For binary compatibility.
+  @deprecated("Use the new implementaion that behaves like orElse", since = "2.2.0")
```

Comment:
Minor pedantry: I would specify the new implementation more explicitly. Maybe something like this:

```suggestion
  @deprecated("use catsEffectSemigroupKForResource2, which behaves more like orElse (or implicitly via SemigroupK[Resource[F, *]])", since = "2.2.0")
```

Ish? Not sure.